web-client/src/app/state/session-xrpl.service.ts

```diff
@@ -423,39 +467,54 @@ export class SessionXrplService {
 
   protected async prepareUnsignedTransaction(
     receiverId: string,
-    amount: xrpl.Payment['Amount']
+    amount: xrpl.Payment['Amount'],
+    sender?: string
   ): Promise<xrpl.Payment> {
     const { wallet } = this.sessionQuery.assumeActiveSession();
+    const senderId = sender
+      ? sender
+      : this.sessionQuery.assumeActiveSession().wallet.xrpl_account.address_base58;
 
     return withLoggedExchange(
       'SessionXrplService.sendFunds: XrplService.createUnsignedPaymentTransaction:',
       async () =>
         await this.xrplService.createUnsignedPaymentTransaction(
-          wallet.xrpl_account.address_base58,
+          senderId,
           receiverId,
           amount
         ),
-      { from: wallet.xrpl_account.address_base58, to: receiverId, amount }
+      { from: senderId, to: receiverId, amount }
     );
   }
 
-  protected async signTransaction(
-    txnUnsigned: xrpl.Transaction
+  protected async signXrplTransaction(
+    txnUnsigned: xrpl.Transaction,
+    public_key_hex?: string,
+    wallet_id?: string,
+    account_pin?: string
   ): Promise<string> {
     const { wallet } = this.sessionQuery.assumeActiveSession();
 
+    const wallet_hex_key = public_key_hex
+      ? public_key_hex
+      : wallet.xrpl_account.public_key_hex
+
     const { txnBeingSigned, bytesToSignEncoded } = txnBeforeSign(
       txnUnsigned,
-      wallet.xrpl_account.public_key_hex
+      wallet_hex_key
+      // wallet.xrpl_account.public_key_hex
     );
 
     const transactionToSign: TransactionToSign = {
       XrplTransaction: {
         transaction_bytes: hexToUint8Array(bytesToSignEncoded),
       },
     };
-    const signed: TransactionSigned = await this.sessionService.signTransaction(
-      transactionToSign
+
+    const signed: TransactionSigned = await this.signTransaction(
```

Comment:
Why did we change `this.sessionService.signTransaction` to `this.signTransaction`? Are they duplicate methods?

Reply:
The problem I have, as a js noob, is that if I import signTransaction from sessionService, within session-xrpl, I get the error "Error in callback TypeError: _this3.sessionService.signTransaction is not a function". Any idea why this is happening? Is it because I'm importing from sessionService which is importing from sessionXrplService?

Reply:
It looks like the circular import requires some refactoring.

Reply:
I don't think we should move the signTransaction method from session to session-xrpl.
All enclave methods is called from session.

Reply:
All methods in session.service are wrappers for enclave methods.
I understand that we refactored it like this to make it work but we're breaking the structure of our code if we refactor it like this.

Reply:
I moved signtransaction back to session.services and autoSendFund to register.page.ts.